File: tools/split-lib-maven-plugin/src/main/java/cn/dev8/util/InputStreamTextReaderThread.java
package cn.dev8.util;

import lombok.Data;
import lombok.SneakyThrows;
import lombok.var;

import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.LinkedList;


@Data
public class InputStreamTextReaderThread  extends Thread  {

    private   InputStream inputStream;
    private   int bufferTime;
    private BufferedReader br = null;


    @SneakyThrows
    public InputStreamTextReaderThread(InputStream  inputStream  , int bufferTime){
        this.inputStream = inputStream;
        this.bufferTime = bufferTime;
        this.br = new BufferedReader(new InputStreamReader(inputStream, "utf-8"), 1024);
    }

    @SneakyThrows
    @Override
    public void run() {
        while (true) {
            try {
                var line = br.readLine();
                if (line.isEmpty()) {
                    break;
                }

                if (line.startsWith("Progress (") && line.contains("/")) {
                    continue;
                }

                this.results .add(line);

                if (done) {
                    break;
                }
            } catch (  Exception e ) {
                error = e;
                break;
            } finally {
                if (done) {
                    br.close();
                }
            }
        }
    }

    private Exception error = null;

    public LinkedList<String> results =  new LinkedList<String>();

    private boolean  done = false;

    @SneakyThrows
    public void  done() {
        this.done = true;
        this.join(bufferTime   * 2);
    }
}

File: ktbase/kotlin/nbcp/model/InputStreamTextReaderThread.kt
package nbcp.model

import java.io.BufferedReader
import java.io.InputStream
import java.io.InputStreamReader

class InputStreamTextReaderThread(var inputStream: InputStream, var bufferTime: Int = 1000) : Thread() {
    private var br = BufferedReader(InputStreamReader(inputStream, "utf-8"), 1024);


    override fun run() {
        while (true) {
            try {
                var line = br.readText()
                if (line.isEmpty()) {
                    break;
                }

                if (line.startsWith("Progress (") && line.contains("/")) {
                    continue;
                }

                this.results += line

                if (done) {
                    break;
                }
            } catch (e: Exception) {
                error = e;
                break;
            } finally {
                if (done) {
                    br.close();
                }
            }
        }
    }

    var error: Exception? = null
        get
        private set

    var results = mutableListOf<String>()
        get
        private set

    private var done = false;
    fun done() {
        this.done = true;
        this.join(bufferTime.toLong() * 2);
    }
}